Added mode to input function arguments

k-w-w opened the pull request.

tensorflower-gardener committed:
In the Grappler arithmetic optimizer, make sure the two new nodes have unique names when hoisting common factors out of aggregates.

Fix a few missed optimization opportunities due to control dependencies.

Fix a bug in counting the number of inputs for trivial aggregate rewriting.

PiperOrigin-RevId: 175639520

tensorflower-gardener committed:
Merged commit includes the following changes:
175703479  by yifeif:

    Internal change.

--
175695370  by A. Unique TensorFlower:

    Implements _shared_embedding_columns and adds some tests.

--
175695349  by A. Unique TensorFlower:

    Implements tf.metrics.true_negatives, adds missing tests and does some cleanup in tf.contrib.metrics.

--

PiperOrigin-RevId: 175703479

alextp committed:
Tape stack in C++ instead of python.

PiperOrigin-RevId: 175704617

tensorflower-gardener committed:
[tpu:profiler] Add matrix unit utilization to TfOpStats.

PiperOrigin-RevId: 175713542

tensorflower-gardener committed:
Adding an optimizer module for preconditioned stochastic gradient langevin dynamics.

PiperOrigin-RevId: 175714379

tensorflower-gardener committed:
Fixed a bug in tensorflow::str_util::CUnescape. Added a str util test that
failed without this change.

The CUnescape did a const_cast to the result string's buffer, which made it write
the same buffer without copying.

PiperOrigin-RevId: 175714391

tensorflower-gardener committed:
Added support for multi-column vocab files to tf.contrib.lookup.index_table_from_file

PiperOrigin-RevId: 175715120

alextp committed:
Removing generator expression in args_to_matching_eager

PiperOrigin-RevId: 175715743

mrry committed:
Do not log a warning when `Rendezvous::Send()` fails in the Send op kernel.

A failing call to `Send()` indicates that the step has been aborted by
a corresponding call to `Rendezvous::StartAbort()`. As a result, the
error logged by `Send()` is not particularly informative, and creates
a non-deterministic amount of extra log spam for each step that fails
as `Send()` calls are being issued. The failure that causes the step to be
aborted is logged separately by the kernel that failed, unless that kernel
deliberately does not log on failure.

In particular, this change reduces log spam when using `Iterator.get_next()`
in a multi-device setting. The `Iterator.get_next()` op deliberately does not
log when an `OutOfRange` error (indicated the end of the dataset) is raised,
because this is common and expected behavior, especially when using an
initializable iterator that is reinitialized at the end of an epoch. Previously,
when running in distributed mode or using a GPU, pending `Send()` calls may
cause unwanted log messages to be printed.

Fixes #12414.

PiperOrigin-RevId: 175716290

tensorflower-gardener committed:
fix a typo in the comment.

PiperOrigin-RevId: 175716586

Commit:
Adding Python3.6 support for nightly binaries.

PiperOrigin-RevId: 175716707

benoitsteiner committed:
Build the function graph only once when inferring shapes for function. This speeds things up quite a bit when ShapeRefiner::RunShapeFn needs more than one pass to compute the shapes.

PiperOrigin-RevId: 175717420

hawkinsp committed:
Remove experimental tpu.outside_all_rewrites() API.

PiperOrigin-RevId: 175718301

tensorflower-gardener committed:
Enabling default extend_export_strategy call to use the base_export_strategy name as this is a common case.

PiperOrigin-RevId: 175719299

MarkDaoust committed:
The link fixer needs these to be all on one line.

PiperOrigin-RevId: 175721637

rohan100jain committed:
Enable prefetching on the resnet50 benchmark for eager.

PiperOrigin-RevId: 175722984

alextp committed:
Clear trace in C instead of python.

PiperOrigin-RevId: 175723846

alextp committed:
Do not create dtype objects in the eager hot path for execution, use enums

PiperOrigin-RevId: 175724282

Commit:
Update tf.keras Dataset, Engine and Layers to the Keras 2.0.9 API.

PiperOrigin-RevId: 175726451

nealwu committed:
Remove wide_n_deep_tutorial.py in tensorflow/examples/learn in favor of wide_deep.py in the TensorFlow official models

PiperOrigin-RevId: 175728483

tensorflower-gardener committed:
Automated g4 rollback of changelist 175571632

PiperOrigin-RevId: 175729221

rjpower committed:
Check for placeholder ops (other than features and labels) during graph validation.

We can provide a more helpful error message at this point than if we detect this at XLA compile time.

PiperOrigin-RevId: 175730190

aselle committed:
Fix speech hotword model links.

tensorflower-gardener committed:
boosted_trees: Finalize the centering bias tree only after centering is done (to make sense).

PiperOrigin-RevId: 175733336

allenlavoie committed:
Make save/restore non-members of tfe.Network. This should make it easier to move
to core.

tfe.Network.save -> tfe.save_network_checkpoint
tfe.Network.restore -> tfe.restore_network_checkpoint

Some minor changes in the restore-on-load logic to make it work as a non-member of Network (particularly in _add_deferred_restoration). The other code changes are trivial, just moving code around.

PiperOrigin-RevId: 175735659

tensorflower-gardener committed:
Switch use of gRPC Generic API to Generic Unary, reducing CQ trips from 4->1

PiperOrigin-RevId: 175736600

tensorflower-gardener committed:
Fix docstring typo: "GraphKey" -> "GraphKeys".

PiperOrigin-RevId: 175736748

Commit:
Add `tf.contrib.bayesflow.layers`, a collection of probabilistic (neural) layers.

PiperOrigin-RevId: 175737756

tensorflower-gardener committed:
Refactors the WALS estimator so that part of the control flow logic happens in the SweepHook. This fixes a bug that causes both input batches (rows and columns) to be fetched during any given sweep.

PiperOrigin-RevId: 175738242

sguada committed:
Make tf.make_template compatible with eager.

PiperOrigin-RevId: 175738521

rryan committed:
Add seed to tf.contrib.layers.dropout.

This allows customizing the dropout seed via arg_scope.

PiperOrigin-RevId: 175740302

annarev committed:
Internal change.

PiperOrigin-RevId: 175740778

Commit:
Use Toggle instead of bool to make the layout optimizer name and usage consistent with other optimizers.

PiperOrigin-RevId: 175743440

tensorflower-gardener committed:
A few profiler improvements.
1. Op view proto copy uses too much memory and time, optimized.
2. Add a hint to use "bazel-bin" instead of "bazel run"
3. Make proto string parsing explicit (seems no longer throwing error)

PiperOrigin-RevId: 175745677

rjpower committed:
Make the definition of summary operations a warning instead of raising an exception.

PiperOrigin-RevId: 175748972

tensorflower-gardener committed:
[TF:XLA] Added tf_xla_cpu_global_jit flag to TF_XLA_FLAGS environment variable to enable global JIT compilation for CPU via SessionOptions. By default, global JIT compilation
for CPU via SessionOptions is disabled. When TF_XLA_FLAGS=--tf_xla_cpu_global_jit is set,
the value of enable_jit_by_default variable in mark_for_compilation_pass.cc is ignored allowing XLA to use JIT compilation for the whole graph according to SessionOptions setting .

Unless tf_xla_cpu_dev_mode is explicitly set via TF_XLA_FLAGS, this code change
should have no effect on Tensorflow or XLA execution.

RELNOTES: n/a
PiperOrigin-RevId: 175754729

hawkinsp committed:
Replace uses of @%ws% with @org_tensorflow in LLVM BUILD file. @%ws% existed to work around a bug in older Bazel versions. The minimum Bazel version no longer has this issue so we can just write @org_tensorflow.

Fixes #14445.

PiperOrigin-RevId: 175756784

tensorflower-gardener committed:
Add head name to tf.contrib.estimator.regression_head metrics

PiperOrigin-RevId: 175757275

Commit:
Update tf.keras to the Keras 2.0.9 API.

PiperOrigin-RevId: 175757949

jpienaar committed:
[TFXLA] Add source node and make GetSwitchCluster more conservative.

PiperOrigin-RevId: 175758538

alextp committed:
python3 fixes

PiperOrigin-RevId: 175758757

k-w-w committed:
Added mode to input function arguments

alextp committed:
Benchmark for tfe_py_execute(identity), for reference with the other identity benchmarks

PiperOrigin-RevId: 175763756

meheffernan committed:
Fix a few issues with HloValue and dataflow analysis identified when debugging the causes for the rollback of the new copy insertion (cl/174423881):

(1) Mark values for deletion during dataflow propagation and delete later, rather than delete immediately. It was possible for a value to be deleted (a phi is optimized away), and still have references to it in the value sets.

(2) Make call/while and root instructions explicit uses of the values which reach them. This subsumes the need for the HloValue::live_out_of_computation_ property which was buggy (which computation is it live-out of?).

(3) Delete unused methods HloValue::RecomputeUses and HloValue::RemovePosition.

PiperOrigin-RevId: 175765613

tensorflower-gardener committed:
Added support for multi-column vocabs to the int-to-string lookup API.

PiperOrigin-RevId: 175767044

tensorflower-gardener committed:
Remove a Done TODO from Svdf op.

PiperOrigin-RevId: 175767193

rjpower committed:
Improve the model comparison utility.

Add session reset, verify input function is consistent and freeze random variables.

PiperOrigin-RevId: 175768806

tensorflower-gardener committed:
Update BUILD visibility.

PiperOrigin-RevId: 175770716

tensorflower-gardener committed:
Remove unused BUILD dependencies

PiperOrigin-RevId: 175772549

Commit:
[XLA:GPU] Mark loads as invariant where appropriate.

If we read a value within an HLO that isn't modified by that HLO, mark
it as invariant in LLVM IR.

LLVM can perform more aggressive optimizations on invariant loads, but I
don't expect this will help much in our case, because XLA already emits
pretty aggressive noalias information on loads and stores.

However, on nvidia GPUs, marking loads as invariant has the additional
benefit of allowing LLVM to lower the load as ld.global.nc (equivalent
to CUDA's __ldg).  This instruction uses a special cache on the GPU, and
it's usually faster than a regular load.

PiperOrigin-RevId: 175774979

tensorflower-gardener committed:
Add a control dependency optimizer to Grappler.

The first two rewrites implemented are:

1. Turn nodes with only control outputs into NoOps, if we know that they are safe to remove. Such nodes can be produced, e.g., by rewrite rules in the arithmetic optimizer.

2. Completely disconnect NoOp nodes with at most 1 input or at most 1 output by rerouting their inputs to their outputs. The restriction on fan-in/fan-out guarantees that we reduce the number of control dependencies in the graph. The two (slightly) non-trivial cases are:

  // Case a)
  //    x --^> +------+                x --^> +---+
  //    y --^> | NoOp | --^> a   ==>   y --^> | a |
  //    ...    |      |                  ...  |   |
  //    z --^> +------+                z --^> +---+
  //
  // Case b)
  //           +------+ --^> a         +---+ --^> a
  //    x --^> | NoOp | --^> b  ==>    | x | --^> b
  //           |      | ...            |   | ...
  //           +------+ --^> c         +---+ --^> c

PiperOrigin-RevId: 175780178

mrry committed:
Do not log an error when `tf.py_func()` raises `StopIteration`.

This reduces log spam when using `tf.py_func()` to wrap a Python generator,
such as when using `tf.data.Dataset.from_generator()`.

PiperOrigin-RevId: 175831781

ispirmustafa reviewed (commented):
Please add a unit test.
Thanks!

mrry committed:
Remove `FunctionDefLibrary` argument from `Device::MaybeRewriteGraph()`.

No existing subclass of `Device` reads that argument, and it incurs a protobuf serialization of the function library each time a new subgraph is created in `DirectSession::Run()`.

PiperOrigin-RevId: 175837162

Commit:
Correct the logging message as (very likely) we will support queue-based input
pipeline.

PiperOrigin-RevId: 175848995

tensorflower-gardener committed:
Update layout_optimizer gradient test to use a variable for the input instead of a constant. Otherwise the dependency optimizer eliminates the Conv2DBackpropInput.

PiperOrigin-RevId: 175851832

Commit:
Move dot-related helpers into dot_op_emitter.cc

This keeps related logic together in a single file.

PiperOrigin-RevId: 175852532

tensorflower-gardener committed:
Add support for Squeeze Op quantization

PiperOrigin-RevId: 175853708

jart committed:
Add graph writer op to contrib/summary

This change also defines a simple SQL data model for tf.GraphDef, which
should move us closer to a world where TensorBoard can render the graph
explorer without having to download the entire thing to the browser, as
that could potentially be hundreds of megabytes.

PiperOrigin-RevId: 175854921

tensorflower-gardener committed:
Don't enable dependency optimizer by default.

PiperOrigin-RevId: 175857095

tensorflower-gardener committed:
Go: Update generated wrapper functions for TensorFlow ops.

PiperOrigin-RevId: 175861269

ebrevdo committed:
TensorArray changes: respect infer_shape argument inside the TensorArray object.

This adds a new attr to the TensorArrayV3 construction op: identical_element_shapes (default False).

If True, then shape for all elements is inferred at runtime when any single element is written.

The Python TensorArray constructor's "infer_shape" argument is piped through to this attribute.  Since it is true by default, this enables runtime element consistency checking on top of the existing graph build time static shape checking.

PiperOrigin-RevId: 175862771

nickdesaulniers committed:
Add test util for setting init_value of SumReduce, ReduceWindow, and SelectAndScatter ops to a Constant 0.0f.

PiperOrigin-RevId: 175864310

nealwu committed:
Add a link to the guide on "Using savedmodel with estimators" to the documentation for export_savedmodel

PiperOrigin-RevId: 175864747

tensorflower-gardener committed:
Adds a SetInvalidOutput method to XlaOpKernelContext.

PiperOrigin-RevId: 175865046

annarev committed:
Internal change.

PiperOrigin-RevId: 175865309

tensorflower-gardener committed:
Update ops-related pbtxt files.

PiperOrigin-RevId: 175866123

tensorflower-gardener committed:
Go: Update generated wrapper functions for TensorFlow ops.

PiperOrigin-RevId: 175867164

alextp committed:
Removes eager memory trace. Has been broken and unrunnable for a while (since EagerTensor went to C).

If we want its functionality we can recover it from the C tape code I believe.

PiperOrigin-RevId: 175869936

fchollet committed:
Final steps for updating tf.keras to the Keras 2.1.1 API.

PiperOrigin-RevId: 175871749

Commit:
Use a static "linker initialized" tensorflow::mutex when possible.

There is no need to use a lazily created tensorflow::mutex since the
tensorflow::LINKER_INITIALIZED constructor is a no-op.
PiperOrigin-RevId: 175874749

mrry committed:
Add `WorkerService.DeleteWorkerSession` method to fix a memory leak.

The new method is the counterpart to `WorkerService.CreateWorkerSession`, and
is called in all cases where worker sessions have been explicitly created (i.e.
when using ClusterSpec propagation).

PiperOrigin-RevId: 175877407

alextp committed:
Fast-path zeros and ones for backprop

PiperOrigin-RevId: 175877719

allenlavoie committed:
Switch tfe.Network to zero-based naming to match Layer variable names.

The plan is that Layer names will transition to match their variable names
(currently Layer names are one-based, but their variable names are
zero-based). Since we can't change variable names, it's better to match
those. Layers added to Networks will follow this convention now.

PiperOrigin-RevId: 175878910

eliben committed:
Make per-platform test disabling mechanism more flexible by allowing regular expressions.

PiperOrigin-RevId: 175879989

tensorflower-gardener committed:
Use optimized reverse_row funciton to flip uint8 images.

In the imagenet preprocessing pipeline, we can always flip the image first then resize it.
It's cheaper to flip a small uint8 image.

PiperOrigin-RevId: 175880318

Artem-B committed:
Improve ptxas-related error handling.

* Don't crash on cubin file cleanup as the file may not have been created.
* Log only one error message if ptxas is not found.

PiperOrigin-RevId: 175882482

tensorflower-gardener committed:
Merged commit includes the following changes:
175895735  by gunan:

    Move IODataType to a new types.proto in preparation for using it
    in ModelFlags

--
175893178  by fchollet:

    Fix multiprocessing issue in keras.utils.data_utils.

--
175891377  by jart:

    Use redundant download URLs and fix protobuf def

    I uncommented all the GitHub URLs that turned out to have consistent
    SHAs. The ones that didn't, I left them commented. We can't re-mirror
    those URLs because doing so would cause past revisions of the
    repository to become broken. So workspace.bzl is going to have to
    evolve back to not having these comments, over time, as we continue to
    upgrade stuff.

--
175889431  by apassos:

    Fixed order for EagerVariableStore.

--
175882680  by A. Unique TensorFlower:

    Move IODataType to a new types.proto in preparation for using it
    in ModelFlags

--

PiperOrigin-RevId: 175895735

tensorflower-gardener committed:
Added ability to specify estimation mode in constructor of KfacOptimizer.  This functions analogously to the estimation_mode argument to FisherEstimator (and is just passed directly to the FisherEstimator that the optimizer constructs).

PiperOrigin-RevId: 175897318

jart committed:
Improve docstring on contrib summary module

PiperOrigin-RevId: 175900586

tensorflower-gardener committed:
Add missing Unref of TensorReference in async CheckNumericsOp.

PiperOrigin-RevId: 175904805

Commit:
Expose an Orc JIT memory mapper registry.

XLA clients can use this registry to inject client-specific behavior into how
Orc JIT's manages virtual memory.

PiperOrigin-RevId: 175905401

Commit:
Fix a build error in windows debug build (#14404)

Required by std::_Debug_range<_FwdIt> in std::lower_bound function.

tensorflower-gardener committed:
Fix MarkDown formatting of code block.

PiperOrigin-RevId: 175908614

allenlavoie committed:
Rename layers.base.Network -> layers.network.GraphNetwork

Splits GraphNetwork out into a new file, moves some shared utility functions to
layers.utils. Should have no functional changes.

PiperOrigin-RevId: 175909000

skye committed:
Refactor Operation.__init__ to create some state after creating _c_op

This change moves around the Operation.__init__ logic to create the
TF_Operation before initializing _outputs and before adding the op to
the control flow context (if any). This is in preparation for creating
Operation objects around TF_Operations indirectly created by the C API
(e.g. ops created by TF_ImportGraphDef).

This also disables running HessianTest with the C API enabled, since
it's broken for now (but will be fixed soon).

PiperOrigin-RevId: 175910443

saeta committed:
Support user-space DNS caching for the GCS filesystem.

In some environments, DNS resolution is unreliable. This change adds an optional userspace caching mechanism to radically reduce the amount of DNS queries sent to upstream resolvers.

PiperOrigin-RevId: 175910642

benoitsteiner committed:
Don't assert when processing invalid dimensions

PiperOrigin-RevId: 175910804

tensorflower-gardener committed:
Hlo parser: support infeed and outfeed.

PiperOrigin-RevId: 175911331

jcastagneri committed:
Fix strides format for data format in contrib.layers.separable convolution2d (#12273)

* Strides must change form if data format is switched

* Fixed stride mismatch on channels first data format for separable conv

* Fixed sepconv2d test and added stride testing

tensorflower-gardener committed:
Cast offsets into int64 to allow for cases where tag_indices is of type int64 (e.g. the output of a call to tf.lookup.*).

PiperOrigin-RevId: 175912063

Commit:
Minor change in VirtualScheduler logging: there's sometimes a difference between device total uptime and the sum of per-op computation time, because uptime includes waiting for channel communications.

PiperOrigin-RevId: 175912780

tensorflower-gardener committed:
Fix data race in tensorflow/core/kernels/map_stage_op.cc, and use tf locking/formatting  conventions.

- Fix data race in which a routine mysteriously releases a lock which was held when
  it was called, and which the caller expects will still be held afterwards.(!)
  See the changes to the routines notify_inserters_if_bounded() and notify_removers().

  The data race was in put() on current_bytes_ near the end, after the call to
  put_complete().  put_complete() calls notify_removers(), which released the
  lock, thus causing the race.  Any reader of the code would have expected that
  the lock would have been held at that point, but it wasn't.

  The same pattern in notify_inserters_if_bounded() didn't cause a race in the
  current code, but is a pitfall for future maintainers, who would not mornally
  expect something called notify_inserters_if_bounded() to mean "and in some
  circumstances, unlock".

  Even in the absense of this bug, and the weird, uncommented spec for those
  routines, it is almost always a bad idea to release the lock before calling a
  condition-variable wakeup.  First, it complicates the code, and makes
  it hard to reason about object deletion if the deletion is predicated by one
  of the signalled conditions.  Second, it's an _unnecessary_ complication:  a
  well-written condition variable implementation will requeue the woken
  thread(s) on the mutex queue, so the expensive part of the wakeup will be
  deferred until the waking thread unlocks anyway.

- Avoid the C++11 unique_lock mechanism that allowed the bug to happen,
  and instead use tensorflow::mutex, tensorflow::mutex_lock, and
  tensorflow::condition_variable.

- Use while-loops instead of lambdas for condition-variable waits, because
  while-loops are much easier to read.

- Fix the routines would_exceed_memory_limit() and is_capacity_full() to
  include their own validity check, so it's not needed at each of their call
  sites.  This improves the readability at the call sites.

- Use annotalysis locking annotations so that other violations can be found
  statically by the compiler.

- Reformat with clang-format.

PiperOrigin-RevId: 175913085

jhseu committed:
Fix the mac build (#14602)

yunxing committed:
Add necessary shape util support for bfloat16

RELNOTES: Add necessary shape util support for bfloat16.
PiperOrigin-RevId: 175914798

tensorflower-gardener committed:
Verify file content before constructing the model

PiperOrigin-RevId: 175914923

allenlavoie committed:
Switch the op naming of tfe.Network to match its variable naming scheme.

e.g. uses my_network_1/dense_1/ consistently rather than also using
my_network_1_1/dense_1/ sometimes.

PiperOrigin-RevId: 175915162

tensorflower-gardener committed:
Fix a bug when printing fusion_kind in hlo_graph_dumper.

PiperOrigin-RevId: 175915347

Commit:
Rename HloToProfileIndex to HloProfileIndexMap

Also fix a typo in a nearby comment.

PiperOrigin-RevId: 175915436

tensorflower-gardener committed:
[XLA] Adding kConditional opcode that represents a conditional HLO instruction.

PiperOrigin-RevId: 175919301

saxenasaurabh committed:
Make MapDataset saveable.
Also integrate functionality of GraphDefBuilderWrapper::AddDatasetWithInputAsList into AddDataset since MapDatasetOp requires 2 inputs of type single tensor and tensor list.

PiperOrigin-RevId: 175921068

tensorflower-gardener committed:
Implements specifying default approximations for layer_collection.

Currently, the default approximation to use for each layer is hard-coded as
a default argument to each registration function. This CL instead specifies
these default values as properties which the user can modify. Additionally,
the user can identify groups of linked parameters that should always use a
specified approximation when registered. This should make it easier for
users to experiment with different approximations.

PiperOrigin-RevId: 175955141

tensorflower-gardener committed:
Add docstring note about .transform() backprop behaviour.

PiperOrigin-RevId: 175955706

tensorflower-gardener committed:
MultiHead adds individual head loss in eval_metric_ops.

PiperOrigin-RevId: 175970818

langmore committed:
linear_operator_test_util.py.  Adding hooks for turning off
placeholders/adjoint in tests.

PiperOrigin-RevId: 175972993

benoitsteiner committed:
Simplify reductions in more cases.

PiperOrigin-RevId: 175975917

mrry committed:
[Speech commands] Add `num_classes=label_count` when constructing the confusion matrix (#14600)

This seems like it should fix an issue where you get unlucky, the batch doesn't contain the largest label, and the returned matrix is smaller than other runs.

brettkoonce committed:
minor spelling tweaks in headers (#14570)

jbedorf committed:
Add missing dependency (#14587)

caisq committed:
contrib/summary: refactor summary_test_util

A logdir may contain files other than summary event files, e.g., checkpoints.
So add a method "events_from_file" to load events from a single file.
The existing "events_from_logdir" method now calls the new method.

PiperOrigin-RevId: 175981886

akshayka committed:
Do not change the default graph in variable_scope when building a function.

When building a TensorFlow function, we need precise control over the default
graph.  This change ensures that, when a function is being built,
variable_scope preserves the default graph.

PiperOrigin-RevId: 175983226

alextp committed:
Ops with no outputs in eager should return None instead of []

PiperOrigin-RevId: 175983704

akshayka committed:
Watch variables when executing defun-compiled functions.

Variables are automatically watched upon creation of a _GraphModeFunction; this
change ensures that the same variables are watched during subsequent executions
of it.

PiperOrigin-RevId: 175984583

tensorflower-gardener committed:
Adds _var_scope_name property to FeatureColumn.

PiperOrigin-RevId: 175986319

Commit:
Fix ci_parameterized_build.sh

tensorflower-gardener committed:
Fixed typo in comment.

PiperOrigin-RevId: 175988884

alextp committed:
Execution callbacks monkey-patch the execute function.

PiperOrigin-RevId: 175990829

tensorflower-gardener committed:
Plumb worker max_wait_secs arguments up to tf.contrib.train.train.

PiperOrigin-RevId: 175991159

tensorflower-gardener committed:
Revise tf.substr so that it returns an empty string when "pos" argument is equal to the string length.

The updated behavior is consistent with strings::substr()

PiperOrigin-RevId: 175992569

Commit:
Make CPU's IrEmitter::hlo_to_profile_idx_ a value

I think the performance advantages of keeping it as a maybe-null pointer are
minimal, and it instead complicates the signature generation code.  For example,
the code to generate calls to __xla_cpu_runtime_ParallelForkJoin is buggy when
hlo_to_profile_idx_ is nullptr today.  This bug isn't visible today because we
always have hlo_to_profile_idx_ as nullptr in JIT mode and in AOT mode we don't
parallelize Hlo operations.

PiperOrigin-RevId: 175993645

tensorflower-gardener committed:
Internal Change

PiperOrigin-RevId: 175997220

tensorflower-gardener committed:
Update ops-related pbtxt files.

PiperOrigin-RevId: 175999357

tensorflower-gardener committed:
Go: Update generated wrapper functions for TensorFlow ops.

PiperOrigin-RevId: 176000517

tensorflower-gardener committed:
Multidimensional sparse float tensors support - changing split criteria, stats accumulator and quantiles op to take the dimension into account.

PiperOrigin-RevId: 176001234

tensorflower-gardener committed:
Update links to point to tflite

PiperOrigin-RevId: 176001760

benoitsteiner committed:
Keep the node map up to date to ensure that nodes won't get deleted by mistake

PiperOrigin-RevId: 176003007

isaprykin committed:
Use the provided optimizer for computing gradients.

Since the user provides an optimizer, gradients should be computed via optimizer.compute_gradients() rather than just the tf.gradients() call.

PiperOrigin-RevId: 176003625

tensorflower-gardener committed:
Tag tensorflow/python:basic_session_run_hooks_test with "notsan".

The test gets intermittent TSan warnings.

PiperOrigin-RevId: 176011109

Commit:
Fix nopip build tag typo in contrib/predictor:test_export_dir.

PiperOrigin-RevId: 176011110

tensorflower-gardener committed:
Hlo parser: allow rank 2 convolution dimension lables.

PiperOrigin-RevId: 176013570

jtkeeling committed:
Correct markdown for code segments in gradients_impl

PiperOrigin-RevId: 176015049

meheffernan committed:
Add methods on TransferManager which transfer to/from device memory specified by ShapedBuffer rather than DeviceMemoryBase. This is part of a broader replacement of DeviceMemoryBase->ShapedBuffer in several XLA interfaces. With this change TransferManager no longer has to allocate memory to transfer tuples to the device. The existing methods using DeviceMemoryBase will be removed in a followup cl.

Various related changes:

* Make the transfer_manager_test an xla_test so that it runs on all the platforms.

* Make several of the TransferManager methods protected.

* Change ScopedShapedBuffer::Allocate to only allocate device memory buffers, and not fill in the tuple index table. The index table is filled in by the transfer manager. This is a cleaner separation of concerns.

PiperOrigin-RevId: 176015628

tensorflower-gardener committed:
[XLA] Make a LOG(FATAL) a bit more useful.

PiperOrigin-RevId: 176019239

tensorflower-gardener committed:
[tf.data] Iterator save and restore patch for Dataset.from_tensors(..) and Dataset.from_tensor_slices(..)

PiperOrigin-RevId: 176019513

tensorflower-gardener committed:
[tf.data] Saveable iterator for TFRecordDatasetOp.

PiperOrigin-RevId: 176020932

Commit:
[XLA:CPU] Add an explicit code path for the entry computation cycle count

While this does change the profile counter entry for the entry computation
during AOT compiles (earlier it would always be some non-null llvm::Value, but
now it can be null), it does not change any observable behavior since
RecordCompleteComputation is a no-op for an empty hlo_to_profile_idx_ map.

PiperOrigin-RevId: 176022629

jhseu committed:
Disable for generated_examples_zip_test in open-source (#14635)

* Disable for OSS

* Buildifier

tensorflower-gardener committed:
Move IODataType to a new types.proto in preparation for using it
in ModelFlags

PiperOrigin-RevId: 176024307

gunan committed:
Disable non-hermetic testcase in gcs_dns_cache_test.

PiperOrigin-RevId: 176027515

tensorflower-gardener committed:
Small (literally) typo in description.

PiperOrigin-RevId: 176031889

miaout17 committed:
iOS Camera example for TensorFlow Lite. (#14591)

The code is derived from the TensorFlow mobile camera example in
tensorflow/tensorflow/examples/ios/camera/.

matslina committed:
typo: s/cesnus/census/ (#14588)

brettkoonce committed:
minor spelling tweaks (#14564)

PinkySan committed:
Exporting tensorflow package for cmake (#14554)

* Exporting Targets

* reverting changes within tf_core_framework.cmake

gunan committed:
Move ptr_util to core/util. (#14647)

* Move ptr_util to core/util.

Break the dependency of core onto XLA.

* Reformat build file.

* Fix the include directive.

qmick committed:
fix #14542 a bug of model_to_dot() (#14553)

samuelzhouhe committed:
Fix the wrong ValurError formatting in convolutional.py (#14551)

Commit:
Removes non-existent link (#14538)

winnietsang committed:
fix broken section links in the document (#14524)

mrry committed:
Portable path parsing in speech_commands/input_data.py (#14519)

* Portable path parsing in speech_commands/input_data.py

Convert some code using a regular expression with `'/'` characters into code that uses `os.path` methods and handles `'\'` characters in Windows paths.

* Move the `.lower()` to the proper spot.

karllessard committed:
Remove obsolete Input interface (#14517)

karllessard committed:
Allow passing a list of shapes as an attribute to an operation (#14644)

facaiy committed:
PrefetchDatasetOp checks buffer_size in c++ side (#14508)

* TST: add test case

* BUG: check buffer_size > 0 on c++ side

taehoonlee committed:
Add missing conv1d in `tf.contrib.layers` (#14513)

taehoonlee committed:
Fix typos (#14516)

boeddeker committed:
fix assert_shallow_structure for dicts (#14499)

* fix assert_shallow_structure for dicts

* fix assert_shallow_structure for dict in util.nest

* fix wrong NestTest (wrong key)

Commit:
use with when calling TFRecordWriter (#14497)

TFRecordWriter supports __enter__ and __exit__ . 
Calling it through with is more pythonic and does cleanup in case something throws an uncaught exception.

dcunhas committed:
Fixed incorrect documentation in tf.contrib.nn.deprecated_flipped_sparse_softmax_cross_entropy_with_logits where it referenced labels rather than logits. (#14485)

k-w-w committed:
Added mode argument to input_fn in tests.

k-w-w commented:
Oops, did something wrong there... I'm going to delete this branch and try again.
